### PR TITLE
[WIP] Add Jukebox model 

### LINKED ISSUE #16870: OpenAI's Jukebox for music generation
### Model description

**[Jukebox](https://openai.com/blog/jukebox/)** is an autoregressive model for music generation published by OpenAI in 2020. It is based on a [hierarchical VQ-VAE](https://arxiv.org/abs/1906.00446) and Scalable Transformers (based on [Sparse Transformers](https://arxiv.org/abs/1904.10509) ) to create long music samples that can be conditioned on Genres, Artists, Timing and Lyrics. 

3 sampling strategies will be made available : 

- Ancestral sampling : tokens are generated in an autoregressive fashion and are then upsampled
- Windowed sampling :  in order to generate long sequences, samples are repeatedly produced from overlapping windows using the previous codes as context. 
- Primed sampling : a continuation of a previous audio is obtained using the VQ-VAE encoding of the audio as initial tokens for the ancestral sampling

The generated tokens are then passed through the VQ-VAE decoder to obtain the final audio. 

The Lyric conditional informations are obtained using a Lyric Transformer model. 

### Open source status

- [X] The model implementation is available
- [X] The model weights are available

### Provide useful links for the implementation

The code is available at https://github.com/openai/jukebox, and weights are available at https://openaipublic.azureedge.net/jukebox/models/. 

Authors : 

- [prafullasd](https://github.com/prafullasd)
- [heewooj](https://github.com/heewooj)
- [jongwook](https://github.com/jongwook)
- [mcleavey](https://github.com/mcleavey)

### PR DESCRIPTION
**Pull request by ArthurZucker**
This is a draft pull request. 

# What does this PR do?
This PR will progressively add the [Jukebox](https://openai.com/blog/jukebox/) model to the hub.
It is linked to [#16870](https://github.com/huggingface/transformers/issues/16870). 

# Currently planned steps (WIP)

- [x]  Create template files with `transformeres-cli add-new-model-like`
- [x] `src/transformers/tokenization_jukebox.py`
- [x]  `src/transformers/test_tokenization_jukebox.py`
- [x]  `src/transformers/configuration_jukebox.py`
- [x]  `src/transformers/modeling_jukebox.py`
- [ ]  `src/transformers/configuration_jukebox.py`
- [ ]  `docs/source/model_doc/jukebox.rst`
- [ ] `src/transformers/tokenization_jukebox_fast.py` (will most probably use WordLevel tokenizer). Also requires to implement a converter function `class JukeboxConverter(Converter):`

### CODE REVIEW

**Comment by ArthurZucker** [comment]
Tokenizer and corresponding test should be done. Lacking some detailed description and also probably something about the arguments in the init that are not data but I don't remember if I should create setters (@patrickvonplaten would love to have your review)

File: tests/jukebox/test_tokenization_jukebox.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+import unittest
+
+# from ..test_tokenization_common import TokenizerTesterMixin
+from transformers import JukeboxTokenizer
+
+
+class JukeBoxIntegrationTest(unittest.TestCase):
+
+    # @slow
+    def test_tokenizer(self):
```
Comment (patrickvonplaten) [inline tests/jukebox/test_tokenization_jukebox.py:34]: Test looks good!

File: tests/jukebox/test_tokenization_jukebox.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            40,
+            77,
+            44,
+            41,
+            27,
+            30,
+        ]
+
+        self.assertTrue(inputs == EXPECTED_OUTPUT)
+        INFINITY = float("inf")
```
[the remark this answers is not in the document]
Comment (ArthurZucker) [reply tests/jukebox/test_tokenization_jukebox.py:72]: The Infinity definition is indeed not required :)

**Comment by patrickvonplaten** [comment]
Cool nice to see much progress here! 

Feel free to also add a file that shows how you compare the original to the current implementation

**Comment by ArthurZucker** [comment]
See followup in #17826

**Pull request closed without merging**